homeassistant/components/squeezebox/browse_media.py

```diff
@@ -268,10 +268,26 @@ def _get_item_thumbnail(
                 item_type, item["id"], track_id
             )
 
-    elif search_type in ["apps", "radios"]:
-        item_thumbnail = player.generate_image_url(item["icon"])
-    if item_thumbnail is None:
-        item_thumbnail = item.get("image_url")  # will not be proxied by HA
+    else:
+        url = None
+        content_type = item_type or "unknown"
+
+        if search_type in ["apps", "radios"]:
+            url = player.generate_image_url(item["icon"])
+        elif image_url := item.get("image_url"):
+            url = image_url
+
+        if url:
+            if internal_request:
+                item_thumbnail = url
+            elif hasattr(entity, "get_synthetic_id_and_cache_url"):
```

Comment:
Why do we need the hasattr guard, is the entity passed in not always a `SqueezeBoxMediaPlayerEntity`?

Reply:
I would have thought so too, but mypy did not like it:

> Error: homeassistant/components/squeezebox/browse_media.py:281: error: "MediaPlayerEntity" has no attribute "get_synthetic_id_and_cache_url"  [attr-defined]

Reply:
Actually.. how would you fix that?
Can `squeezebox/browse_media` import `SqueezeBoxMediaPlayerEntity` from `squeezebox/media_player` even though `squeezebox/media_player` already imports stuff from `squeezebox/browse_media`?